src/compas_ags/diagrams/forcediagram.py

```diff
@@ -301,27 +319,50 @@ def constraints_from_dual(self, tol=10e-4):
         ForceDiagram is modified in place.
         """
         edge_index = self.dual.edge_index()
+        _edges = list(self.edges())
         ordered_edges = self.ordered_edges(self.dual)
+        edges_orient = []
 
-        for edge in self.edges_where_dual({'is_ind': True}):
+        for edge in self.edges_where_dual({'is_ind': True}):  # Fix vertices of dual independent edge
             self.vertices_attribute('is_fixed', True, keys=[*edge])
-            self.edge_attribute(edge, 'has_fixed_orientation', True)
+            edges_orient.append(edge)
 
-        for edge in self.edges_where_dual({'has_fixed_orientation': True}):
-            self.edge_attribute(edge, 'has_fixed_orientation', True)
+        for edge in self.edges_where_dual({'is_load': True}):  # If loads are orthogonal the force dual edge gets constrained
```

Comment:
why?

Reply:
should load edges not always be constrained?

Reply:
well... now we can for example assign target forces to these loads, then they change their magnitude, so loads in general are just constrained to its original orientation. I constrain only the extremities of the force edge of the independents, which in theory you should not change the force by means of "target force".